### PR TITLE
IS-1021: DictDB iteration causes infinite loop

### PR DESCRIPTION
**Pull request by leeheonseung**
officially we do not support iteration on DictDB
so add raise Exception code

references
https://www.python.org/dev/peps/pep-0234/
https://docs.python.org/3/reference/datamodel.html#object.__getitem__

### CODE REVIEW

File: tests/integrate_test/samples/sample_scores/sample_dict_db/sample_dict_db.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    @external
+    def create_item(self, key: str, value: int) -> None:
+        self._dict[key] = value
+
+    @external(readonly=True)
+    def get_items(self) -> list:
+        items = []
+        for item in self._dict:
+            print("11")
```
Comment (eunsoo-icon) [inline tests/integrate_test/samples/sample_scores/sample_dict_db/sample_dict_db.py]: remove debug message

File: iconservice/iconscore/icon_container_db.py [hunk 2]
```diff
@@ -216,6 +216,9 @@ def __remove(self, key: K) -> None:
             raise InvalidContainerAccessException('DictDB depth mismatch')
         self._db.delete(get_encoded_key(key))
 
+    def __iter__(self):
+        raise InvalidContainerAccessException("Not Supported on DictDB")
```
Comment (eunsoo-icon) [inline iconservice/iconscore/icon_container_db.py]: What is not supported? Need more Info.

File: tests/icon_score/test_icon_container_db.py [hunk 3]
```diff
@@ -339,3 +339,33 @@ def test_container_util(self):
...
+        if key > self._limit:
+            raise IndexError(key)
+
+
+class TestIterObj(TestOnlyGetItemObj):
+    def __iter__(self):
+        pass
+
+
+class TestContainerDB(unittest.TestCase):
```
Comment (eunsoo-icon) [inline tests/icon_score/test_icon_container_db.py]: What is the purpose of tests? Please let me know.

File: tests/integrate_test/samples/sample_scores/sample_dict_db/sample_dict_db.py (hunk 1 above)
Comment (leeheonseung) [reply tests/integrate_test/samples/sample_scores/sample_dict_db/sample_dict_db.py]: fixed

File: iconservice/iconscore/icon_container_db.py (hunk 2 above)
Comment (leeheonseung) [reply iconservice/iconscore/icon_container_db.py]: fixed

File: tests/icon_score/test_icon_container_db.py (hunk 3 above)
Comment (leeheonseung) [reply tests/icon_score/test_icon_container_db.py]: This test created a situation where iterable with getitem and it could take an infinite loop.

**Pull request merged**